awx/main/tasks/facts.py

```diff
@@ -62,7 +62,8 @@ def start_fact_cache(hosts, destination, log_data, timeout=None, inventory_id=No
 
 
 def raw_update_hosts(host_list):
-    Host.objects.bulk_update(host_list, ['ansible_facts', 'ansible_facts_modified'])
+    host_list = sorted(host_list, key=lambda host: host.id)
+    Host.objects.bulk_update(host_list, ['ansible_facts', 'ansible_facts_modified'], batch_size=100)
 
 
 def update_hosts(host_list, max_tries=5):
```

Comment:
Later in this file, you can see a loop:

```python
for host in hosts_cached:
```

The calls to `update_hosts` are _manually batched_, and I would expect that causes the same issue as Django's `batch_size`. This is using a batch size inside of a batch. You might as well get rid of the `batch_size` kwarg.

As you do the main loop over `hosts_cached`, I expect the hosts need to be cached there.